### PR TITLE
PR to have AGFusion have a soft error when no fusions are found

### PR DESCRIPTION
**Pull request by ksinghal28**
As described in issue https://github.com/griffithlab/analysis-wdls/issues/68 AGFusion currently throws an error if no fusions are detected at the RNASeq Star Fusion detect step, resulting in the pipeline completely erroring out.

This PR modifies the check_data function in parsers.py to have a warning message stating no fusions were found instead of completely erroring out. It also makes an empty agfusion_results.zip file that is needed for the later steps.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ENV ensembl_version=105
 
 # Download latest AGFusion from source and install with pip
 WORKDIR /usr/local/bin
-RUN git clone https://github.com/genome/AGFusion.git
+RUN git clone https://github.com/ksinghal28/AGFusion.git
```
Comment (tmooney) [inline Dockerfile]: This change shouldn't be included in the PR.

File: agfusion/parsers.py [hunk 2]
```diff
@@ -24,7 +26,13 @@ def __next__(self):
 
     def _check_data(self):
         if len(self.fusions) == 0:
-            self.logger.error("Read 0 fusions from the file! Exiting...")
+            #self.logger.error("Read 0 fusions from the file! Exiting...")
+            if not os.path.exists("agfusion_results"):
+                os.mkdir("agfusion_results")
+            with zipfile.ZipFile("/cromwell_root/agfusion_results.zip", mode="w") as archive:
```
Comment (tmooney) [inline agfusion/parsers.py]: If AGFusion is run outside of cromwell this seems likely to break.  But I don't think we should be making this file here.  AGFusion doesn't make a ZIP file of its results normally.  If one is needed, that should be created by the workflow step, just as it currently is in the case where results exist.

Comment (malachig) [reply agfusion/parsers.py]: I agree that hardcoding an expectation of`/cromwell_root/ ...` is probably not what we want 

Two comments.
1.) As far as I know this fork of AGfusion only exists for us to run workflows
2.) I'm not sure we would follow your last suggestion though. I think by the time we know we need to create an empty result, AGfusion has already died, and this kills the workflow...

Comment (malachig) [reply agfusion/parsers.py]: Is there a way that we can catch the failure of AGfusion here?:
https://github.com/griffithlab/analysis-wdls/blob/018f5c0721252a57c22435a68123f611612fb7f1/definitions/tools/agfusion.wdl#L18-L25 

And if that command did not succeed, instead of going into the output dir and zipping up the results, we just create an empty result right there in that block in the WDL?

Comment (malachig) [reply agfusion/parsers.py]: That has its own problems in my mind.  We don't want to do this for *any* failure of AGfusion.  We wanted it to be somehow specific to this case where AGfusion tells us. I found nothing to annotate.

Comment (malachig) [reply agfusion/parsers.py]: What if we let AGfusion create an empty results dir at this point but do NOT create the zip file?  Let that part happen in the WDL as is it would normally?

Comment (malachig) [reply agfusion/parsers.py]: And instead of creating an empty folder with hard coded name, we use the output dir name that passed into AGfusion as a param.

Comment (malachig) [reply agfusion/parsers.py]: It seems like maybe an output directory is created?  Or does agfusion die before it gets to this point...
https://github.com/genome/AGFusion/blob/c839a2769f49050eeff71c13ae8653345d7ef91b/agfusion/cli.py#L151-L152

Comment (tmooney) [reply agfusion/parsers.py]: > 2.) I'm not sure we would follow your last suggestion though. I think by the time we know we need to create an empty result, AGfusion has already died, and this kills the workflow...

I agree that we can have the tool make an empty result as needed, but the ZIP file specifically is never created by AGFusion. It is created here normally:
https://github.com/griffithlab/analysis-wdls/blob/018f5c0721252a57c22435a68123f611612fb7f1/definitions/tools/agfusion.wdl#L24
It doesn't make sense to instead make it here in the case that there are no fusions.  That's not a responsibility of this tool.

**Comment by ksinghal28** [comment]
Hi, I just tested my most recent commit and it seems to be working as expected. 
@tmooney thanks for suggesting I make a text file in the directory before zipping it! 

Please let me know if there's anything else I can change, or cleanup!

**Comment by tmooney** [comment]
I'd remove the new but commented-out code entirely.  Besides that be sure to not change the `Dockerfile`--we'll still want it to point to this repo once the changes are merged!  Otherwise looks good to me!

File: agfusion/parsers.py [hunk 3]
```diff
@@ -2,8 +2,10 @@
 Parses output files from fusion-finding algorithms
 """
 
+import os
 import re
 import csv
+import zipfile
```
Comment (tmooney) [inline agfusion/parsers.py]: We can probably remove this one too!

Comment (ksinghal28) [reply agfusion/parsers.py]: Just took it out, thanks for catching that!

**Review by tmooney — APPROVED** [review]
Looks good to me!
